### PR TITLE
[Deps] eo+eolian+efl

### PR DESCRIPTION
**Pull request by joaoantoniocardoso**
Branching from `devs/felipealmeida/native-windows/eolian-api` (no more `EFL_EXTERN` thing needed here).

What is building:
- eolian
  - [x] lib
  - [x] bin
  - [x] tests
- eo
  - [x] lib
  - [x] tests
- efl
  - [x] lib
  - [x] tests: not enabled, _depends on ecore_

### CODE REVIEW

File: src/lib/eo/eo_ptr_indirection.c [hunk 1]
```diff
@@ -39,9 +39,9 @@ _eo_obj_pointer_invalid(const Eo_Id obj_id,
    const char *reason = "This ID has probably been deleted";
    char tbuf[128];
    if (obj_id & MASK_CLASS_TAG) type = "class";
-   if (thread != _efl_object_main_thread)
+   if (!eina_thread_equal(thread, _efl_object_main_thread))
      {
-        snprintf(tbuf, sizeof(tbuf), "%p", (void *)thread);
```
Comment (cauebs) [inline src/lib/eo/eo_ptr_indirection.c:44]: Maybe we could do something like:
```C
   Eina_Thread thread = eina_thread_self();
   Eina_ThreadId thread_id = eina_thread_self_id();
   ...
   if (!eina_thread_equal(thread, _efl_object_main_thread))
     {
        snprintf(tbuf, sizeof(tbuf), "%ld", (long) thread_id);
...
```
I'm just not sure if that cast will work fine with a `DWORD`, `int`, `pid_t` and `size_t`

Comment (Coquinho) [reply src/lib/eo/eo_ptr_indirection.c:44]: Is this line even a problem? I mean, it only prints the pointer right? Does it matter how the data is defined?

Comment (cauebs) [reply src/lib/eo/eo_ptr_indirection.c:44]: It's used for logging. I think it's important to print out a value that helps developers with debugging.

Comment (JPTIZ) [reply src/lib/eo/eo_ptr_indirection.c:44]: I aggree with getting the thread_id explicitly. It's more righteous with expecting the ID to be shown, and not _insert abstract data-type here_.

Comment (joaoantoniocardoso) [reply src/lib/eo/eo_ptr_indirection.c:44]: Done: 06208c2

File: meson.build [hunk 2]
```diff
@@ -490,7 +490,7 @@ if sys_windows
     'edje',
     'eet',
     'eeze',
-    'efl',
+    #'efl',
```
Comment (Coquinho) [inline meson.build:493]: Why not just removing? same to eo and eolian

Comment (joaoantoniocardoso) [reply meson.build]: done: 74aac71

**Comment by Coquinho** [comment]
@joaoantoniocardoso, there are some conflicts here.

**Comment by joaoantoniocardoso** [comment]
Now it is ready to me merged.

File: src/lib/eina/eina_types.h [hunk 3]
```diff
@@ -285,6 +285,8 @@
 # endif
 # define EINA_MALLOC
 # define EINA_PURE
+# define EINA_HOT
+# define EINA_COLD
```
Comment (Coquinho) [inline src/lib/eina/eina_types.h:289]: What are those for?

File: src/lib/eldbus/Eldbus.h [hunk 4]
```diff
@@ -144,7 +144,7 @@ typedef struct _Eldbus_Version
    int revision; /**< git revision (0 if a proper release or the git revision number Eldbus is built from) */
 } Eldbus_Version;
 
-EAPI extern const Eldbus_Version * eldbus_version; /**< Global Eldbus_Version object */
```
Comment (Coquinho) [inline src/lib/eldbus/Eldbus.h:147]: `Eldbus` isn't compiled at windows, do we really need to configure `EXTERN` here?

**Comment by joaoantoniocardoso** [comment]
> Wouldn't it make more sense to only apply the `EXTERN` thing only at `eo`, `eolina`, and `efl` on this PR?

It makes sense, but now I think that other five or six branches are using the benefits of dealing with all of externs at this one. Changing it now I think unnecessary... And later perhaps all of this extern mods will be reverted as they are not the perfect solution but allow us to continue to advance for now.

File: src/lib/eina/eina_types.h (hunk 3 above)
Comment (joaoantoniocardoso) [reply src/lib/eina/eina_types.h:289]: see b0a102a

File: src/lib/elput/Elput.h [hunk 5]
```diff
@@ -125,17 +125,17 @@ typedef struct _Elput_Event_Switch
 } Elput_Event_Switch;
 
 
-EAPI extern int ELPUT_EVENT_SEAT_CAPS;
```
Comment (Coquinho) [inline src/lib/elput/Elput.h:128]: `Elput` isn't compiled at windows, do we really need to configure `EXTERN` here?

File: src/lib/eina/eina_types.h (hunk 3 above)
Comment (cauebs) [reply src/lib/eina/eina_types.h:289]: Those came from my branch. They were needed but not defined for Windows. Under GCC they are defined to use GCC-specific attributes (`__attribute__(hot)` and `__attribute__(cold)`), for optimization purposes.

File: src/lib/evil/unposix/strings.h [hunk 6]
```diff
@@ -3,10 +3,10 @@
...
-#define strncasecmp _strnicmp
-#define strcasecmp _stricmp
-#define strncasecmp_l _strnicmp_l
-#define strcasecmp_l _strncmp_l
+// #define strncasecmp _strnicmp
+// #define strcasecmp _stricmp
+// #define strncasecmp_l _strnicmp_l
+// #define strcasecmp_l _strncmp_l
 
-inline int ffs(int i);
```
Comment (Coquinho) [inline src/lib/evil/unposix/strings.h:11]: What made those not important?

Comment (joaoantoniocardoso) [reply src/lib/evil/unposix/strings.h:11]: already defined at `src/lib/evil/evil_strings.h`

Comment (Coquinho) [reply src/lib/evil/unposix/strings.h:11]: hmm, okay, so why not delete it?

File: src/lib/elput/Elput.h (hunk 5 above)
Comment (joaoantoniocardoso) [reply src/lib/elput/Elput.h]: You are right. Elput and Eldbus changes are really not necessary as they will never be part of win32 build. 12cbffc5a4

File: src/lib/evil/unposix/strings.h (hunk 6 above)
Comment (joaoantoniocardoso) [reply src/lib/evil/unposix/strings.h]: done: b733c67

File: src/lib/eldbus/Eldbus.h (hunk 4 above)
Comment (joaoantoniocardoso) [reply src/lib/eldbus/Eldbus.h]: done: 12cbffc

**Comment by cauebs** [comment]
> > Wouldn't it make more sense to only apply the `EXTERN` thing only at `eo`, `eolina`, and `efl` on this PR?
> 
> It makes sense, but now I think that other five or six branches are using the benefits of dealing with all of externs at this one. Changing it now I think unnecessary... And later perhaps all of this extern mods will be reverted as they are not the perfect solution but allow us to continue to advance for now.

Just be careful, you're missing includes in files in which you're making the change to `EFL_EXTERN`. They build on Windows probably because of other includes (e.g. evil), but don't under Linux. One example is `src/lib/eina/eina_log.h`

**Comment by Coquinho** [comment]
Got a little confused here, why `28df9c364d37eca9f4fec4b51a782acb5ec77806` and `706e001efb53860b901c4bf14492daf9673a9e01` are needed?

**Comment by Coquinho** [comment]
Hey, just to let you know, Linux didn't like that `EXTERN`:
```
[8/2653] Compiling C object 'src/lib/eina/803ee99@@eina@sha/eina_cpu.c.o'.
FAILED: src/lib/eina/803ee99@@eina@sha/eina_cpu.c.o 
cc -Isrc/lib/eina/803ee99@@eina@sha -Isrc/lib/eina -I../src/lib/eina -I. -I../ -I/usr/include/valgrind -fdiagnostics-color=always -pipe -D_FILE_OFFSET_BITS=64 -fvisibility=hidden -Wall -Wpointer-arith -Wunused-parameter -Wsign-compare -Wno-missing-field-initializers -DHAVE_CONFIG_H=1 -D_GNU_SOURCE=1 -DEFL_BETA_API_SUPPORT=1 -DELM_INTERNAL_API_ARGESFSDFEFC=1 -fPIC -pthread '-DPACKAGE_DATA_DIR="/usr/local/share/eina"' -DNEED_RUN_IN_TREE=1 -DEFL_BUILD=1 -MD -MQ 'src/lib/eina/803ee99@@eina@sha/eina_cpu.c.o' -MF 'src/lib/eina/803ee99@@eina@sha/eina_cpu.c.o.d' -o 'src/lib/eina/803ee99@@eina@sha/eina_cpu.c.o' -c ../src/lib/eina/eina_cpu.c
In file included from ../src/lib/eina/eina_cpu.c:49:
../src/lib/eina/eina_log.h:233:16: error: expected ‘;’ before ‘int’
  233 | EAPI EFL_EXTERN int EINA_LOG_DOMAIN_GLOBAL;
      |                ^~~~
      |                ;
../src/lib/eina/eina_log.h:982:16: error: expected ‘;’ before ‘const’
  982 | EAPI EFL_EXTERN const char *_eina_log_state_init;
      |                ^~~~~~
      |                ;
../src/lib/eina/eina_log.h:984:16: error: expected ‘;’ before ‘const’
  984 | EAPI EFL_EXTERN const char *_eina_log_state_shutdown;
      |                ^~~~~~
      |                ;
```
Compiled with `meson build/ -Dbindings=cxx,mono`

File: src/benchmarks/eo/class_simple.c [hunk 7]
```diff
@@ -7,8 +7,8 @@
 
 #define MY_CLASS SIMPLE_CLASS
 
-EOAPI const Efl_Event_Description _SIMPLE_FOO = EFL_EVENT_DESCRIPTION("foo");
-EOAPI const Efl_Event_Description _SIMPLE_BAR = EFL_EVENT_DESCRIPTION("bar");
+const Efl_Event_Description _SIMPLE_FOO = EFL_EVENT_DESCRIPTION("foo");
```
Comment (cauebs) [inline src/benchmarks/eo/class_simple.c:10]: Why not EO_API?

Comment (felipealmeida) [reply src/benchmarks/eo/class_simple.c:10]: Por que class_simple.c é um executável, não uma biblioteca. Ele não precisa exportar ou importar o simbolo que ele mesmo define e usa.

**Comment by Coquinho** [comment]
Hey, @joaoantoniocardoso sorry about that, but I think this needs some rebase with #171 and #173.
I needed it to see more on ecore, so I've done it here [devs/coquinho/break-in-peaces/eo+eolian+efl](https://github.com/expertisesolutions/efl/tree/devs/coquinho/break-in-peaces/eo+eolian+efl) if you want to take it :)

File: src/lib/efl/efl_api.h [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+#  endif
+# else
+#  define EFL_API
+# endif
+# define EFL_API_WEAK
+#else
+# ifdef __GNUC__
+#  if __GNUC__ >= 4
+#   define EFL_API __attribute__ ((visibility("default")))
+#   define EFL_API_WEAK __attribute__ ((weak))
```
[the remark this answers is not in the document]
Comment (Coquinho) [reply src/lib/efl/efl_api.h:23]: It is remaining of #173 actually, maybe move this topic to it?

**Pull request merged**